src/App.tsx

```diff
@@ -1,19 +1,138 @@
 import React from 'react';
 import './App.css';
-
+import { TodoList } from './components/TodoList/TodoList';
+import todosFromServer from './api/todos';
 import users from './api/users';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const newTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
+
+type Props = {};
+
+type State = {
+  todos: Todo[],
+  newTitle: string,
+  selectedId: string,
+  hasTitleError: boolean,
+  hasUserSelectedError: boolean,
 };
 
+class App extends React.PureComponent<Props, State> {
+  state: State = {
+    todos: [...newTodos],
+    newTitle: '',
+    selectedId: '0',
+    hasTitleError: false,
+    hasUserSelectedError: false,
+  };
+
+  handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    const regexp = /^[а-яА-Яa-zA-Z0-9\s,.'Ёё]+$/;
+
+    if (regexp.test(value)) {
+      this.setState((state) => ({
+        ...state,
+        newTitle: value,
+        hasTitleError: false,
+      }));
+    } else if (value.length > 1) {
+      this.setState((state) => ({
+        newTitle: state.newTitle,
+      }));
+    } else {
+      this.setState(() => ({
+        newTitle: '',
+      }));
+    }
```

Comment:
It's the unnecessary condition, cause if it's empty title, you can use previous condtion